### PR TITLE
fix: environment variables

### PR DESCRIPTION
**Pull request by Freyskeyd**
# Description

This PR is fixing an issue with the environment variables where the `source .env` was overwriting context environment variables, leading to wrong tests setup.

## PR Checklist:

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added or updated tests that comprehensively prove my change is effective or that my feature works

### CODE REVIEW

File: .github/workflows/frontend:erc20-messaging.yml [hunk 1]
```diff
@@ -7,6 +7,7 @@ on:
   workflow_dispatch:
     inputs:
       e2e-test-type:
+        required: true
```
Comment (sebastiendan) [inline .github/workflows/frontend:erc20-messaging.yml]: I don't think that's needed, right?

File: .github/workflows/topos:integration-tests.yml [hunk 2]
```diff
@@ -31,7 +31,8 @@ jobs:
     runs-on: ubuntu-latest-16-core
     environment: devnet-1
     env:
-      INFRA_REF: ${{ inputs.local-erc20-messaging-infra-ref || needs.infra-metadata.outputs.latest-release }}
+      # INFRA_REF: ${{ inputs.local-erc20-messaging-infra-ref || needs.infra-metadata.outputs.latest-release }}
+      INFRA_REF: "chore/debug-ci-1"
```
Comment (sebastiendan) [inline .github/workflows/topos:integration-tests.yml]: This needs to be removed

File: .github/workflows/frontend:erc20-messaging.yml (hunk 1 above)
Comment (Freyskeyd) [reply .github/workflows/frontend:erc20-messaging.yml]: we need it in order to have the default value set during `branch` event.

File: .github/workflows/topos:integration-tests.yml (hunk 2 above)
Comment (dvdplm) [reply .github/workflows/topos:integration-tests.yml]: @Freyskeyd Is this here temporarily to test it with a pending PR? Is there an alternative to this?

File: .github/workflows/topos:integration-tests.yml [hunk 3]
```diff
@@ -67,18 +68,23 @@ jobs:
           cd infra &&
           npm install
 
+      - name: Setup environment
+        run: |
+          cd infra
+          cat .env | grep -v '^\s*$\|^\s*\#' | xargs -I {} echo "{}" >> $GITHUB_ENV
```
Comment (dvdplm) [inline .github/workflows/topos:integration-tests.yml:73]: This is… not great? Seems like a hack to me.

File: .github/workflows/frontend:erc20-messaging.yml (hunk 1 above)
Comment (dvdplm) [reply .github/workflows/frontend:erc20-messaging.yml]: Can you add a comment to that effect?

File: .github/workflows/topos:integration-tests.yml (hunk 3 above)
Comment (Freyskeyd) [reply .github/workflows/topos:integration-tests.yml:73]: Do you have a better solution?

File: .github/workflows/frontend:erc20-messaging.yml (hunk 1 above)
Comment (sebastiendan) [reply .github/workflows/frontend:erc20-messaging.yml]: @Freyskeyd Isn't [this change](https://github.com/topos-protocol/e2e-tests/pull/32#discussion_r1444417959) that I had proposed initially what you really want here? How adding a `required` in `workflow_dispatch`'s inputs can impact what is used by the `pull_request` event? (I assume that's the event you were referring to).

And I'm realizing that that proposed change was indeed required.

Comment (Freyskeyd) [reply .github/workflows/frontend:erc20-messaging.yml]: yes indeed we can maybe simplify this

File: .github/workflows/frontend:erc20-messaging.yml [hunk 4]
```diff
@@ -52,14 +53,16 @@ jobs:
     runs-on: ubuntu-latest-16-core
     environment: devnet-1
     env:
-      INFRA_REF: ${{ inputs.local-erc20-messaging-infra-ref || needs.infra-metadata.outputs.latest-release }}
+      # INFRA_REF: ${{ inputs.local-erc20-messaging-infra-ref || needs.infra-metadata.outputs.latest-release }}
+      INFRA_REF: "chore/debug-ci-1"
```
Comment (atanmarko) [inline .github/workflows/frontend:erc20-messaging.yml]: Why this one? Use manual tests to check that branch.

File: .github/workflows/topos:integration-tests.yml (hunk 2 above)
Comment (atanmarko) [reply .github/workflows/topos:integration-tests.yml]: @dvdplm  This is the alternative
image

File: .github/workflows/topos:integration-tests.yml [hunk 5]
```diff
@@ -67,10 +68,17 @@ jobs:
...
+          echo "LOCAL_ERC20_HOME=${PWD}" >> $GITHUB_ENV
+          echo "TOPOS_MESSAGING_PROTOCOL_CONTRACTS_VERSION=${{ env.CONTRACTS_DOCKER_TAG }}" >> $GITHUB_ENV
+          echo "TOPOS_VERSION=${{ env.TOPOS_DOCKER_TAG }}" >> $GITHUB_ENV
+
       - name: Start local ERC20 messaging infra containers
         run: >
           cd infra &&
-          source .env &&
```
Comment (atanmarko) [inline .github/workflows/topos:integration-tests.yml:73]: This was a bug in the test indeed overwriting reference set when the test is trigered

**Review by atanmarko — APPROVED** [review]

**Review by JDawg287 — APPROVED** [review]

**Pull request merged**